lib/validate.js

```diff
@@ -83,8 +83,15 @@ module.exports = function validateOpts (opts, cbPassedIn) {
   }
 
   if (typeof opts.duration === 'string') {
-    if (/[a-zA-Z]/.exec(opts.duration)) opts.duration = timestring(opts.duration)
-    else opts.duration = Number(opts.duration.trim())
+    if (/[a-zA-Z]/.exec(opts.duration)) {
+      try {
+        opts.duration = timestring(opts.duration)
+      } catch (error) {
+        return error
+      }
```

Comment:
If `timestring` was invalid then the error was being swallowed. It is now returned as per other errors.